src/mainwindow.cpp

```diff
@@ -977,7 +969,9 @@ void ConvertJsonToStringMap(const QJsonObject& valJson, const QStringList& field
         }
         else if (val.isString())
         {
-            nameValues[field] = "\"" + val.toString().replace("\n", " ").replace("\"", "\"\"") + "\"";
+            QString strVal = val.toString();
+            strVal.truncate(32000); // Excel cannot handle more than ~32K chars.
+            nameValues[field] = "\"" % strVal.replace("\n", " ").replace("\"", "\"\"") % "\"";
```

Comment:
Added the escape and restore newlines feature to QueryInfo.twb.
Also updated it with 10.1, as that's the version first to support right join in TDE.